### PR TITLE
play: use full workflow ID when re-invoking

### PR DESCRIPTION
**Pull request by oliver-sanders**
* The Cylc CLI can "infer" run numbers e.g. `cylc play foo` might run `foo/run1`.
* This makes monitoring difficult because you can't see what workflow run a Cylc process is running.
* So when/if we re-invoke the `cylc play` command on a remote host we use the full ID (including the run number) not the abbreviated version.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: tests/functional/restart/44-reinvoke.t [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+export REQUIRE_PLATFORM='loc:remote fs:shared runner:background'
+. "$(dirname "$0")/test_header"
+set_test_number 8
```
Comment (wxtim) [inline tests/functional/restart/44-reinvoke.t:20]: I make it 3: Fails Locally.

**Review by wxtim — COMMENTED** [review]
Doesn't pass correctly on my local machine - simple issue.

Otherwise looks reasonable.

File: cylc/flow/scheduler_cli.py [hunk 2]
```diff
@@ -360,24 +360,39 @@ def scheduler_cli(options: 'Values', workflow_id: str) -> None:
...
-def _distribute(host):
-    """Re-invoke this command on a different host if requested."""
+    """
     # Check whether a run host is explicitly specified, else select one.
     if not host:
         host = select_workflow_host()[0]
     if is_remote_host(host):
-        cmd = _protect_remote_cmd_args(sys.argv[1:])
+        # Protect command args from second shell interpretation
+        cmd = list(map(quote, sys.argv[1:]))
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/scheduler_cli.py:381]: This line is tested (see the new test), however, that test doesn't run on GHA.

File: tests/functional/restart/44-reinvoke.t (hunk 1 above)
Comment (oliver-sanders) [reply tests/functional/restart/44-reinvoke.t:20]: Give me a call tomorrow, will take a look in.

**Review by datamel — APPROVED** [review]
Checked out and tested locally. 
No problems spotted from me. Thanks @oliver-sanders.

**Review by wxtim — APPROVED** [review]

**Pull request merged**